Assets/Scripts/GameManager.cs

```diff
@@ -1,25 +1,36 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
 
 public class GameManager : MonoBehaviour
 {
     [SerializeField] SceneLoader sceneLoader;
+    [SerializeField] UIManager uiManager;
+
+    DateTime startTime = DateTime.Now;
 
     // Start is called before the first frame update
     void Start()
     {
         sceneLoader = FindObjectOfType<SceneLoader>();
+        uiManager = FindObjectOfType<UIManager>();
     }
 
     // Update is called once per frame
     void Update()
     {
-        
+        TimeSpan elapsedTime = DateTime.Now - startTime;
+        uiManager.UpdateElapsedTime(elapsedTime.ToString(@"mm\:ss\:ff"));
     }
 
     public void WinGame()
     {
         sceneLoader.LoadWinScreen();
     }
+
+    public void UpdatePoints(int points)
+    {
+        uiManager.UpdatePoints(points);

```

Comment:
Just wanted to highlight this particular addition - I'm not sure if it makes sense for the PlayerSupervisor to pass the new point total to the GameManager just for the GameManager to send it straight to the UIManager. I set this up to match how the structure outline in figure 1 of our project plan, but could see just having the PlayerSupervisor send this directly to the UIManager instead.

Reply:
That should work fine for the single-player mode. I'm not sure if it'll still work as easily when there are multiple PlayerSupervisors talking to the UI manager later, but we'll see about that when two-player mode gets added.